src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -204,15 +199,13 @@ public BitArray(BitArray bits)
         ** Exceptions: ArgumentOutOfRangeException if index < 0 or
         **             index >= GetLength().
         =========================================================================*/
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public bool Get(int index)
         {
-            if ((uint)index >= (uint)Length)
-            {
-                throw new ArgumentOutOfRangeException(nameof(index), index, SR.ArgumentOutOfRange_Index);
-            }
+            if ((uint)index >= (uint)m_length)
+                ThrowArgumentOutOfRangeException(index);
 
-            int elementIndex = Div32Rem(index, out int extraBits);
-            return (m_array[elementIndex] & (1 << extraBits)) != 0;
+            return (m_array[index >> 5] & (1 << index)) != 0;
```

Comment:
Instead of `1 << index` did you mean `1 << (index & 31)` ?

Reply:
C# takes care of that 
> If the type of x is int or uint, the shift count is given by the low-order five bits of the second operand. That is, the shift count is computed from count & 0x1F

Reply:
It might be worth a comment